src/python/pants/option/subsystem.py

```diff
@@ -45,9 +75,32 @@ class Subsystem(metaclass=ABCMeta):
 
     _scope_name_re = re.compile(r"^(?:[a-z0-9_])+(?:-(?:[a-z0-9_])+)*$")
 
+    class EnvironmentAware(metaclass=ABCMeta):
+        subsystem: ClassVar[type[Subsystem]]
+        options: OptionValueContainer
+        env_tgt: EnvironmentTarget
+
+        def __getattribute__(self, __name: str) -> Any:
+            """Fetch the option from the runtime environment, or defer to the global default."""
+            from pants.core.util_rules.environments import get_option
+
+            val = super().__getattribute__(__name)
+            if __name == "options":
+                return val
+
+            try:
+                if __name in self.options:
+                    return get_option(__name, self)
```

Comment:
This name is fairly ambiguous imo in the context of `subsystem.py`. Given it's a helper function only used by use, I think verbosity is worth the clarity `get_option_from_env_or_subsystem`

Reply:
Agreed.

Reply:
Renamed as `resolve_environment_sensitive_option`